action/messageHubFeedWeb.js

```diff
@@ -0,0 +1,192 @@
+            }
+        }, function (authError) {
+            console.log(`authError: ${JSON.stringify(authError)}`);
+            return Promise.reject( 'Could not authenticate with Message Hub. Please check your credentials.' );
+        });
+}
+
+function validateParameters(rawParams) {
+    var promise = new Promise(function (resolve, reject) {
+        var validatedParams = {};
+
+        // topic
+        if (rawParams.topic && rawParams.topic.length > 0) {
+            validatedParams.topic = rawParams.topic;
+        } else {
+            reject( "You must supply a 'topic' parameter." );
+            return;
+        }
+
+        // kafka_brokers_sasl
+        if (rawParams.kafka_brokers_sasl) {
+            validatedParams.brokers = common.validateBrokerParam(rawParams.kafka_brokers_sasl);
+            if (!validatedParams.brokers) {
+                reject( "You must supply a 'kafka_brokers_sasl' parameter as an array of Message Hub brokers." );
+                return;
+            }
+        } else {
+            reject( "You must supply a 'kafka_brokers_sasl' parameter as an array of Message Hub brokers." );
+            return;
+        }
+
+        // user
+        if (rawParams.user) {
+            validatedParams.username = rawParams.user;
+        } else {
+            reject( "You must supply a 'user' parameter to authenticate with Message Hub." );
+            return;
+        }
+
+        // password
+        if (rawParams.password) {
+            validatedParams.password = rawParams.password;
+        } else {
+            reject( "You must supply a 'password' parameter to authenticate with Message Hub." );
+            return;
+        }
+
+        // kafka_admin_url
+        if (rawParams.kafka_admin_url) {
+            validatedParams.kafka_admin_url = rawParams.kafka_admin_url;
+        } else {
+            reject( "You must supply a 'kafka_admin_url' parameter." );
+            return;
+        }
+
+        validatedParams.isJSONData = common.getBooleanFromArgs(rawParams, 'isJSONData');
+        validatedParams.isBinaryValue = common.getBooleanFromArgs(rawParams, 'isBinaryValue');
+
+        if (validatedParams.isJSONData && validatedParams.isBinaryValue) {
+            reject( 'isJSONData and isBinaryValue cannot both be enabled.' );
+            return;
+        }
+
+        // now that everything else is valid, let's add these
+        validatedParams.isMessageHub = true;
+        validatedParams.isBinaryKey = common.getBooleanFromArgs(rawParams, 'isBinaryKey');
+        validatedParams.endpoint = common.massageEndpointParam(rawParams.endpoint);
+        validatedParams.authKey = rawParams.authKey;
+
+        const uuid = require('uuid');
+        validatedParams.uuid = uuid.v4();
+
+        console.log(`I made it this far! ${JSON.stringify(validatedParams, null, 2)}`);
+
+        common.resolveTriggerDetails(validatedParams.authKey, validatedParams.endpoint, rawParams.triggerName)
```

Comment:
The whole point of this is to make sure we have the correct namespace. In this action, the namespace env var will be the action owner's (e.g. whisk.system), not the namespace for the trigger being created. However, @jasonpet points out that the correct namespace _is_ available in the non-web feed action, and could be simply passed into the web action as a parameter value. Brilliant!